pkg/agents/team4/finalAgent/treaty.go

```diff
@@ -0,0 +1,210 @@
+package team4EvoAgent
+
+import (
+	"math"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+/*------------------------OTHER AGENTS TREATY PROPOSAL HANDLING------------------------*/
+
+// CONDITION OF AVALIABLE FOOD
+// If available food <= critical amount * 2 then we reject.
+// Reject a treaty if the conditionop of available food is < or  <= because we don't want to be in a position of loss in any case
+// Reject all treaties where avalaible - request leave food (both amount and percentage) <= critical amount *2.
+
+// CONDITION OF HP
+// Reject all treaties that have condition HP <= critcal * 2 cause we ourselves are desperate at that point
+// Reject all treaties that involve condition HP < or < = because we don't want to be in a position of loss in any case
+
+// CONDITION OF FLOOR
+// Reject all treaties with the conditionop of FLOOR having > or >= because we don't want to be in a position of loss in any case
+
+// Reject all treaties with duration > max days in critical
+// If request is leave food amount or leave percent food and we are asked to leave 100% of the food then reject cause we would need to eat.
+
+func (a *CustomAgentEvo) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treaty := msg.Treaty()
+
+	// Switch Case for rejecting treaties
+	switch {
+	case len(a.ActiveTreaties()) > 0: // If one treaty already signed then we reject any other incoming ones
+		fallthrough
+	case treaty.Request() == messages.Inform: // No idea what inform actually does/how to use it
```

Comment:
Prefer moving these comments to above each `case` so it's more readable. Likewise for the rest of the switch statement, but ultimately up to you